### PR TITLE
NO-ISSUE: 3 bug fixes related to resourceVersion

### PR DESCRIPTION
**Pull request by avishayt**

### CODE REVIEW

File: internal/service/device.go [hunk 1]
```diff
@@ -241,6 +241,7 @@ func (h *ServiceHandler) PatchDevice(ctx context.Context, request server.PatchDe
 	}
 
 	common.NilOutManagedObjectMetaProperties(&newObj.Metadata)
+	newObj.Metadata.ResourceVersion = nil
```
Comment (ori-amizur) [inline internal/service/device.go:244]: Aren't we missing the point of resource version? Don't we want to avoid updating  if someone else updated the resource ?

File: internal/tasks/fleet_rollout.go [hunk 2]
```diff
@@ -216,3 +219,27 @@ func (f FleetRolloutsLogic) getDeviceConfig(device *api.Device, templateVersion
...
+	for i := 0; i < 10; i++ {
+		if device.Metadata.Owner == nil || *device.Metadata.Owner != f.owner {
+			return fmt.Errorf("device owner changed, skipping rollout")
+		}
+
+		device.Spec = newDeviceSpec
+		_, _, err = f.devStore.CreateOrUpdate(ctx, f.resourceRef.OrgID, device, nil, false, f.callbackManager.DeviceUpdatedCallback)
+		if err != nil {
+			if errors.Is(err, flterrors.ErrResourceVersionConflict) {
+				device, err = f.devStore.Get(ctx, f.resourceRef.OrgID, *device.Metadata.Name)
```
Comment (ori-amizur) [inline internal/tasks/fleet_rollout.go:235]: Ignore the returned error?

File: internal/store/device.go [hunk 3]
```diff
@@ -185,9 +193,11 @@ func (s *DeviceStore) Get(ctx context.Context, orgId uuid.UUID, name string) (*a
 
 func (s *DeviceStore) createDevice(device *model.Device) (bool, error) {
 	device.Generation = lo.ToPtr[int64](1)
+	originalResourceVersion := device.ResourceVersion
 	device.ResourceVersion = lo.ToPtr[int64](1)
 	if result := s.db.Create(device); result.Error != nil {
 		err := flterrors.ErrorFromGormError(result.Error)
+		device.ResourceVersion = originalResourceVersion
```
Comment (ori-amizur) [inline internal/store/device.go:200]: I don't understand why is this needed.  We create the device from the `api.Device` during every retry iteration.

File: internal/service/device.go (hunk 1 above)
Comment (avishayt) [reply internal/service/device.go:244]: No, as I mentioned in the commit message, this is a PATCH.  The user didn't specify the resourceVersion - they just want to replace some field in the DB, regardless of what's currently there.  From the kubernetes docs:

> Instead of sending a PUT request, the client can send an instruction to the API server to patch an existing resource. A patch is typically appropriate if the change that the client wants to make isn't conditional on the existing data. Clients that need effective detection of lost updates should consider making their request conditional on the existing resourceVersion (either HTTP PUT or HTTP PATCH), and then handle any retries that are needed in case there is a conflict.

File: internal/tasks/fleet_rollout.go (hunk 2 above)
Comment (avishayt) [reply internal/tasks/fleet_rollout.go:235]: Good call, fixed

File: internal/service/device.go (hunk 1 above)
Comment (ori-amizur) [reply internal/service/device.go:244]: OK

File: internal/store/device.go (hunk 3 above)
Comment (avishayt) [reply internal/store/device.go]: You're right.  I was fixing the tests and I guess some failure led to me thinking this was an issue.  Reverted.

**Review by ori-amizur — APPROVED** [review]
